iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/mapreduce/IcebergInputFormat.java

```diff
@@ -261,6 +268,13 @@ public boolean nextKeyValue() throws IOException {
       while (true) {
         if (currentIterator.hasNext()) {
           current = currentIterator.next();
+          Configuration conf = context.getConfiguration();
+          if (HiveIcebergStorageHandler.isDelete(conf, conf.get(Catalogs.NAME))) {
+            if (current instanceof GenericRecord) {
+              PositionDeleteInfo pdi = IcebergAcidUtil.parsePositionDeleteInfoFromRecord((GenericRecord) current);
+              PositionDeleteInfo.serializeIntoConf(conf, pdi);
```

Comment:
Why do we change the pdi in the conf for every record?

Reply:
Mostly because the row_position changes for every record. If the record reader has multiple tasks (i.e. reading multiple files), then the file_path or even the spec_id/partition_struct could change too

Reply:
My issue is that the PDI contains the position info too. That should not change anything.
My main issue here is that we do the parsing / serialization for every record which will be resource intensive

Reply:
As discussed, I'll change it to store the 4 fields separately into the config instead of serializing the object.